component/common/config/kubernetes.go

```diff
@@ -0,0 +1,87 @@
+package config
```

Comment:
Should these types go into a new package at `component/common/config/kubernetes`? I don't think `ClientArguments` is very descriptive with the package it's been moved to.